query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala

```diff
@@ -367,23 +367,30 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
     log.debug(s"Attempting to map query where expression '${qeOpt}' to rollup ${r}")
 
     (qeOpt, r.where) match {
-      // don't support rollups with where clauses yet.  To do so, we need to validate that r.where
-      // is also contained in q.where.
-      case (_, Some(re)) => None
+      // To allow rollups with where clauses, validate that r.where is contained in q.where (top level and).
+      case (Some(qe), Some(re)) if (topLevelAndContain(re, qe)) =>
+        val strippedQe = stripExprInTopLevelAnd(re, qe)
+        rewriteExpr(strippedQe, r, rollupColIdx).map(Some(_))
+      case (_, Some(_)) =>
+        None
       // no where on query or rollup, so good!  No work to do.
       case (None, None) => Some(None)
       // have a where on query so try to map recursively
       case (Some(qe), None) => rewriteExpr(qe, r, rollupColIdx).map(Some(_))
     }
   }
 
-  def rewriteHaving(qeOpt: Option[Expr], r: Anal, rollupColIdx: Map[Expr, Int]): Option[Having] = {
+  def rewriteHaving(qeOpt: Option[Expr], qgs: Seq[Expr], r: Anal, rollupColIdx: Map[Expr, Int]): Option[Having] = {
```

Comment:
Can you make the name `qgs` a little more explicit to make it obvious that it is group bys?